platform/mellanox/rules.mk

```diff
@@ -39,7 +39,9 @@ $(SYNCD)_DEPENDS += $(MLNX_SAI)
 $(SYNCD)_UNINSTALLS += $(MLNX_SAI)
 
 ifeq ($(ENABLE_SYNCD_RPC),y)
-$(SYNCD)_DEPENDS += $(LIBSAITHRIFT_DEV)
+# Remove the libthrift_0.11.0 dependency injected by rules/syncd.mk
+$(SYNCD)_DEPENDS := $(filter-out $(LIBTHRIFT_DEV),$($(SYNCD)_DEPENDS))
```

Comment:
This seems like a workaround of the build architecture that is in place.

I believe the correct way to achieve this effect is to...
1. Make it such that doing `make configure PLATFORM=mellanox` sets the `SAITHRIFT_V2` to `y`
2. Modify `syncd.mk` to check `SAITHRIFT_V2` when deciding which thrift version to use (since this is global)

Reply:
To save a variable during this step "make configure" and use it later, the only way is to save the variable to a file as we do for platform and arch or 2) I could add check in slave.mk to check for ENABLE_RPC and PLATFORM and set the SAITHRIFT_V2 variable to y. 

I believe both of this will have more questions from the forum. If we have to go by the existing standard, i think we'll have to use SONIC_SAITHRIFT_V2 var.  

And if we don't have to to use that variable i don't see a way without this workaround of overriding it in our rules.mk since rules/*.mk are read first and then the platform specific rules.mk.

Reply:
Updated